### PR TITLE
[MISC] Bugfixes

### PR DESCRIPTION
**Pull request by dragomirp**
Fixing bugs noticed during switching the VM charm to templates.

### CODE REVIEW

File: tests/integration/relations/test_db.py [hunk 1]
```diff
@@ -198,15 +197,3 @@ async def test_extensions_blocking(ops_test: OpsTest) -> None:
         raise_on_blocked=False,
         timeout=3000,
     )
-
-
-@pytest.mark.group(1)
-@pytest.mark.unstable
-async def test_relation_with_openldap(ops_test: OpsTest):
```
Comment (dragomirp) [inline tests/integration/relations/test_db.py:205]: The image for the charm cannot be downloaded.

File: src/relations/db.py [hunk 2]
```diff
@@ -237,7 +237,7 @@ def _on_relation_joined(self, join_event: RelationJoinedEvent):
             return
 
         dbs = self.charm.generate_relation_databases()
-        dbs[join_event.relation.id] = {"name": database, "legacy": True}
+        dbs[str(join_event.relation.id)] = {"name": database, "legacy": True}
```
Comment (dragomirp) [inline src/relations/db.py:240]: JSON keys are string

File: src/charm.py [hunk 3]
```diff
@@ -644,22 +646,20 @@ def generate_relation_databases(self) -> Dict[str, Dict[str, Union[str, bool]]]:
         """Generates a mapping between relation and database and sets it in the app databag."""
         if not self.unit.is_leader():
             return {}
-        if dbs := self.get_relation_databases():
-            return dbs
```
Comment (dragomirp) [inline src/charm.py:648]: Regenerating should be fast enough and if the mapping is wrong for whatever reason, always using the cache will never fix it.

File: src/relations/backend_database.py [hunk 4]
```diff
@@ -335,7 +335,6 @@ def _on_relation_departed(self, event: RelationDepartedEvent):
             return
 
         self.postgres.delete_user(self.auth_user)
-        self.charm.peers.remove_user(self.auth_user)
```
Comment (dragomirp) [inline src/relations/backend_database.py:338]: Doesn't exist any more

File: tests/integration/relations/test_db.py (hunk 1 above)
Comment (marceloneppel) [reply tests/integration/relations/test_db.py:205]: Which image? I deployed OpenLDAP through `juju deploy openldap-charmers-openldap --channel edge` and got it in the `Waiting for database relation` state.

**Review by marceloneppel — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Nice stabilisation!

File: tests/integration/relations/test_db.py (hunk 1 above)
Comment (dragomirp) [reply tests/integration/relations/test_db.py:205]: Looking deeper, it looks like this is the podspec charm and the pull failure seems to originate from [the image](https://git.launchpad.net/charm-k8s-openldap/tree/src/charm.py#n55) pulled by the podspec.

**Pull request merged**